src/ext/java/org/opentripplanner/ext/transmodelapi/mapping/ViaRequestMapper.java

```diff
@@ -0,0 +1,75 @@
+package org.opentripplanner.ext.transmodelapi.mapping;
+
+import graphql.schema.DataFetchingEnvironment;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.opentripplanner.ext.transmodelapi.TransmodelGraphQLUtils;
+import org.opentripplanner.ext.transmodelapi.TransmodelRequestContext;
+import org.opentripplanner.routing.api.request.RouteRequest;
+import org.opentripplanner.routing.api.request.RouteViaRequest;
+import org.opentripplanner.routing.api.request.ViaLocation;
+import org.opentripplanner.routing.api.request.request.JourneyRequest;
+import org.opentripplanner.standalone.api.OtpServerRequestContext;
+
+public class ViaRequestMapper {
+
+  /**
+   * Create a RouteViaRequest from the input fields of the viaTrip query arguments.
+   */
+  public static RouteViaRequest createRouteViaRequest(DataFetchingEnvironment environment) {
+    TransmodelRequestContext context = environment.getContext();
+    OtpServerRequestContext serverContext = context.getServerContext();
+    RouteRequest request = serverContext.defaultRouteRequest();
+
+    List<ViaLocation> vias =
+      ((List<Map<String, Object>>) environment.getArgument("via")).stream()
+        .map(viaLocation ->
+          new ViaLocation(
+            GenericLocationMapper.toGenericLocation(viaLocation),
+            false,
+            (Duration) viaLocation.get("minSlack"),
+            (Duration) viaLocation.get("maxSlack")
+          )
+        )
+        .toList();
+
+    List<JourneyRequest> requests = environment.containsArgument("requests")
```

Comment:
I think this should be "segments"

Reply:
Fixed, I'm looking forward to getting to use the generated types in order to not have keep working with these kind of strings.